src/main/java/bob/TaskList.java

```diff
@@ -0,0 +1,149 @@
+package bob;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import bob.exception.MissingDatesException;
+import bob.exception.MissingTaskException;
+import bob.exception.WrongInputException;
+import bob.task.Deadline;
+import bob.task.Event;
+import bob.task.Task;
+import bob.task.TaskType;
+import bob.task.Todo;
+
+/**
+ * Contains list of tasks, and operations that alter it.
+ */
+public class TaskList {
+
+    public ArrayList<Task> lst;
+
+    public TaskList() {
+        this.lst = new ArrayList<>();
+    }
+    public TaskList(ArrayList<Task> lst) {
+        this.lst = lst;
+    }
+
+    /**
+     * Generates the appropriate type of Task based on user input
+     * Throws exceptions due to incorrect user input
+     * @param description is of the form e.g. "event read /from 2pm /to 4pm"
+     * @return the relevant Task
+     * @throws WrongInputException for unrecognised input.
+     * @throws MissingTaskException when task name is missing.
+     * @throws MissingDatesException when start and end of Event is missing.
+     */
+    public Task generateTask(String description)
+            throws WrongInputException, MissingTaskException,
+            MissingDatesException, DateTimeParseException {
+
+        // Split by the first " " into type, and task details
+        String[] task = description.split(" ", 2);
```

Comment:
Good job on following the coding standard that the array specifier is attached to the type.